packages/integration/lib/src/bloc/comment/handler/get_comment_handler.dart

```diff
@@ -0,0 +1,98 @@
+part of '../comment_bloc.dart';
+
+Future<void> _getCommentHandler(LMFeedGetCommentsEvent event, emit) async {
+  if (event.page == 1) {
+    emit(LMFeedGetCommentLoadingState());
+  } else {
+    emit(LMFeedGetCommentPaginationLoadingState());
+  }
+  final (post, comments) = await fetchCommentListWithPage(
+    event.page,
+    event.postId,
+    event.pageSize,
+  );
+  if (post != null && comments != null) {
+    emit(LMFeedGetCommentSuccessState(post: post, comments: comments, page: event.page));
+  } else {
+    emit(LMFeedGetCommentErrorState(error: 'Failed to fetch comments'));
+  }
+}
+
+Future<(LMPostViewData?, List<LMCommentViewData>?)> fetchCommentListWithPage(
+    int page, String postId, int commentListPageSize) async {
+  PostDetailRequest postDetailRequest = (PostDetailRequestBuilder()
+        ..postId(postId)
+        ..pageSize(commentListPageSize)
+        ..page(page))
+      .build();
+
+  final PostDetailResponse response =
+      await LMFeedCore.client.getPostDetails(postDetailRequest);
+
+  if (response.success) {
```

Comment:
We are checking for failure only in case of deletion of comment or reply, as we have emitted the DelteSuccessState before calling the actual api to hide it localy . and add the comment in case of failure of api call.